src/rust/engine/src/lib.rs

```diff
@@ -539,6 +542,48 @@ pub extern "C" fn lease_files_in_graph(scheduler_ptr: *mut Scheduler) {
   });
 }
 
+#[no_mangle]
+pub extern "C" fn capture_snapshot(
+  scheduler_ptr: *mut Scheduler,
+  raw_root_path: *const raw::c_char,
+  path_globs_value: Value,
+) -> PyResult {
+  let root_path = unsafe { CStr::from_ptr(raw_root_path).to_string_lossy().into_owned() };
+  with_scheduler(scheduler_ptr, |scheduler| {
+    capture_snapshot_inner(scheduler, root_path, path_globs_value).into()
+  })
+}
+
+fn capture_snapshot_inner(
```

Comment:
Keeping this slimmer and making this an `impl Scheduler` method would be preferable I think.

Reply:
Done. Slightly ugly because I can't use `?`s in a function returning `PyResult`, but probably a reasonable separation